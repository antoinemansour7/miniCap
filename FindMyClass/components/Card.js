// Card.js
import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const Card = ({ iconName, title }) => {
  return (
    <TouchableOpacity style={styles.card}>
      <Icon name={iconName} size={40} color="#9B1B30" style={styles.icon} />
      <Text style={styles.title}>{title}</Text> 
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#F5F5F5', // Grayscale background for the cards
    width: 150,
    height: 150,
    margin: 10,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 5, // Adds shadow for Android
  },
  icon: {
    marginBottom: 10,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#333',
  },
});

export default Card;
