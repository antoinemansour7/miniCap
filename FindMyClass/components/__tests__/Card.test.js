import React from 'react';
import { render } from '@testing-library/react-native';
import Card from '../Card';
import Icon from 'react-native-vector-icons/Ionicons';

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

describe('Card Component', () => {
  it('renders correctly with given props', () => {
    const { getByText, getByTestId } = render(<Card iconName="home" title="Home" />);

    // Check if the title is displayed
    expect(getByText('Home')).toBeTruthy();

    // Check if the icon is rendered
    expect(getByTestId('card-icon')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<Card iconName="home" title="Home" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});