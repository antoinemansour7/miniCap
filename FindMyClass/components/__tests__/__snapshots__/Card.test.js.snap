// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "borderRadius": 10,
      "elevation": 5,
      "height": 150,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": 150,
    }
  }
>
  <Icon
    color="#9B1B30"
    name="home"
    size={40}
    style={
      {
        "marginBottom": 10,
      }
    }
  />
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Home
  </Text>
</View>
`;
